system/ui/lib/inputbox.py

```diff
@@ -97,24 +132,37 @@ def render(self, rect, color=rl.BLACK, border_color=rl.DARKGRAY, text_color=rl.W
       if len(display_text) > 0 and self._cursor_position > 0:
         cursor_x += rl.measure_text_ex(font, display_text[: self._cursor_position], font_size, 0).x
 
+      # Apply text offset to cursor position
+      cursor_x -= self._text_offset
+
       cursor_height = font_size + 4
       cursor_y = rect.y + rect.height / 2 - cursor_height / 2
-      rl.draw_line(int(cursor_x), int(cursor_y), int(cursor_x), int(cursor_y + cursor_height), rl.LIGHTGRAY)
+      rl.draw_line(int(cursor_x), int(cursor_y), int(cursor_x), int(cursor_y + cursor_height), rl.WHITE)
+
+    rl.end_scissor_mode()
 
   def _handle_mouse_input(self, rect, font_size):
     """Handle mouse clicks to position cursor."""
     mouse_pos = rl.get_mouse_position()
     if rl.is_mouse_button_pressed(rl.MOUSE_LEFT_BUTTON) and rl.check_collision_point_rec(mouse_pos, rect):
       # Calculate cursor position from click
       if len(self._input_text) > 0:
-        text_width = rl.measure_text_ex(gui_app.font(), self._input_text, font_size, 0).x
-        text_pos_x = rect.x + 10
-
-        if mouse_pos.x - text_pos_x > text_width:
-          self.set_cursor_position(len(self._input_text))
-        else:
-          click_ratio = (mouse_pos.x - text_pos_x) / text_width
-          self.set_cursor_position(int(len(self._input_text) * click_ratio))
+        font = gui_app.font()
+        display_text = "*" * len(self._input_text) if self._password_mode else self._input_text
+
+        # Find the closest character position to the click
+        relative_x = mouse_pos.x - (rect.x + 10) + self._text_offset
+        best_pos = 0
+        min_distance = float('inf')
+
+        for i in range(len(self._input_text) + 1):
```

Comment:
This could probably be optimised instead of searching the entire string, but not likely to cause issues 🙂